### DISCUSSION TITLE
python support

### DISCUSSION THREAD

**Discussion by gmnamra** [OpenImageIO/oiio#2803] [Q&A]
On macOS 10.15.7 

pip3 --version
pip 20.3.3 from /Volumes/medvedev/Users/arman/opt/anaconda3/envs/py39-demo/lib/python3.9/site-packages/pip (python 3.9)

 pip3 install openimageio
ERROR: Could not find a version that satisfies the requirement openimageio
ERROR: No matching distribution found for openimageio

**Comment by lgritz** [comment]
This project isn't responsible for whatever goes on inside pip. I'm not sure who maintains that.

**Reply by lgritz** [reply]
Or are you trying to say "I wish there was a pip-installable openimageio, but there isn't, can you be responsible for that?"

**Comment by Correct-Syntax** [comment]
@lgritz ,

Speaking of a pip-installable openimageio...would you all do one? This would make the usage of openimageio for python projects expand a lot. Then it would make the time you all have spent with the python api a lot more worthwhile. :)

There seems to already be a project that did create pip wheels via Azure DevOps, but the author seems to not be interested in continuing to update for recent versions of oiio: [Github Repository](https://github.com/fredrikaverpil/oiio-python) There also seemed to be some [concerns](https://github.com/fredrikaverpil/oiio-python/issues/12) about the fact that that project wasn't "official" so it would need oiio developer support/agreement to put it on PyPI anyway.

I've forked it and tried to get it working but been having problems on the Azure DevOps end....

*Oops...sorry if this isn't the place to put this...*

**Reply by lgritz** [reply]
I'm all for it, but it's certainly not something I've ever done and I don't really know anything about the process about how to make a package available for installation with pip.

This needs a champion to step forward and volunteer to do it -- figure out exactly what is necessary, submit PRs to change or add anything we need in the OIIO repo, communicate with the PyPI folks as needed, commit to doing the release steps every time we do an OIIO release (generally the 1st of every month) at least for the first several times, and write clear documentation so that any bozo can step in and do the pip things thereafter.

**Reply by Correct-Syntax** [reply]
@lgritz 

> I'm all for it, but it's certainly not something I've ever done and I don't really know anything about the process about how to make a package available for installation with pip.

It's fairly simple. 

Basically you create a *setup.py* file with all the meta info (package title, links, etc, etc...), then create the wheel with ``python3 setup.py sdist bdist_wheel``. Once that's done, it just a matter of uploading to your PyPI  account with ``python3 -m twine upload --repository mypypipackagename dist/*`` That's the basic gist. There may be more required for oiio in specific, but I don't know.

> This needs a champion to step forward and volunteer to do it -- figure out exactly what is necessary, submit PRs to change or add anything we need in the OIIO repo, communicate with the PyPI folks as needed, commit to doing the release steps every time we do an OIIO release (generally the 1st of every month) at least for the first several times, and write clear documentation so that any bozo can step in and do the pip things thereafter.

I am no champion and unfortunately I can't commit to doing anything as I don't have a truly reliable source of internet (currently), but I really want/need oiio for my own projects so I have an incentive to do so.

I can't "sign a contract" or anything, but as long as your forgiving if I am unable to update when the releases come out and I have someone else to figure the Azure DevOps stuff (which I have been unable to get working), then *I would do my best to manage this for the first few times and write some docs as you mentioned.*

Question: I've used Github Actions, but I am not sure how Azure Devops is different. Do you know if there is any difference or could this just be run with Github Actions instead of Azure?

Thanks.

**Reply by lgritz** [reply]
I haven't used Azure Devops. I would imagine that there is a way to do it from GHA by now.

Like I said, if somebody fully figures this out and implements it, we would be happy to accept the PRs, and then if it's basically automated and with clear instructions, we can make it a part of our regular release schedule.

**Comment by BryceGattis** [comment]
@lgritz Just checking in on this, OIIO doesn't have a pip-installable package yet right?

**Comment by lgritz** [comment]
Correct, @BryceGattis, sorry. I believe people are working on it, but I'm not sure exactly how close they are.

**Reply by maxlund** [reply]
Any updates?

**Reply by bhack** [reply]
https://github.com/AcademySoftwareFoundation/OpenImageIO/pull/4428
